clippy_lints/src/utils/internal_lints.rs

```diff
@@ -207,3 +218,31 @@ impl<'a, 'tcx: 'a> Visitor<'tcx> for LintCollector<'a, 'tcx> {
         NestedVisitorMap::All(&self.cx.tcx.hir)
     }
 }
+
+
+
+pub struct PreferFxHash;
+
+impl LintPass for PreferFxHash {
+    fn get_lints(&self) -> LintArray {
+        lint_array!(PREFER_FX_HASH)
+    }
+}
+
+impl EarlyLintPass for PreferFxHash {
+    fn check_ident(&mut self, cx: &EarlyContext, ident: Ident) {
+        if ident.to_string().contains("HashMap") {
```

Comment:
`if` is an expression in Rust, so you can just use it to decide on the message without all the other code duplicated.